x-pack/plugins/observability_solution/apm/server/routes/errors/get_error_groups/get_error_group_main_statistics.ts

```diff
@@ -157,15 +169,30 @@ export async function getErrorGroupMainStatistics({
 
   const errorGroups =
     response.aggregations?.error_groups.buckets.map((bucket) => {
+      const errorSource =
+        'error' in bucket.sample.hits.hits[0]._source
+          ? bucket.sample.hits.hits[0]._source
+          : undefined;
+
+      const event = unflattenKnownApmEventFields(bucket.sample.hits.hits[0].fields, requiredFields);
+
+      const mergedEvent = {
```

Comment:
Same, I think we should use `fields` instead of `_source`

Reply:
I don't know if we reached an agreement on that topic.

What I understood is that APM integration had `exception` as array, while OTel Collector integration will just have the fields as `error.exception.handled`, and others...

I thought both types of `exception` could come from `fields`.

Could you bring some light here @gregkalapos ?

Reply:
Let's check the length, if there is more than one exception, we need to use _source, in other cases we can use `fields`.

Reply:
Is our UI capable of showing more than 1 exception per error?

Reply:
Presumably yes. But more importantly, without using `_source` you'll mix properties of multiple exceptions with each other. E.g. the following:

```json
{
  "exception": [
    {
      "handled": false,
      "type": "unhandled_type"
    },
    {
      "handled": true,
      "type": "handled_type"
    }
  ]
}
```

could result in:
```json
{
  "exception.handled": [ true, false ],
  "exception.type": [ "unhandled_type", "handled_type"]
}
```

I think `fields` also dedupes so the problem is even bigger in practice.